xhr.bs

```diff
@@ -1635,16 +1635,23 @@ interface FormData {
 moved to the HTML Standard. [[HTML]]
 
 <p>The
-<dfn id=dom-formdata constructor for=FormData lt="FormData(form)"><code>new FormData(<var>form</var>)</code></dfn>
+<dfn id=dom-formdata constructor for=FormData lt="FormData(form)"><code>new FormData(<var>form</var>, <var>submitter</var>)</code></dfn>
 constructor steps are:
 
 <ol>
  <li>
   <p>If <var>form</var> is given, then:
 
   <ol>
+   <li>
+    <p>If <var>submitter</var> is not null, then:
+    <ol>
+      <li><p>If <var>submitter</var> is not a submit button, then throw a {{TypeError}}.
+      <li><p>If <var>submitter</var>'s form owner is not <var>form</var>, then throw a "{{NotFoundError!!exception}}" {{DOMException}}.
+    </ol>
```

Comment:
```suggestion
    <p>If <var>submitter</var> is non-null, then:

    <ol>
     <li><p>If <var>submitter</var> is not a <a>submit button</a>, then <a>throw</a> a 
     {{TypeError}}.

     <li><p>If <var>submitter</var>'s <a>form owner</a> is not <var>form</var>, then <a>throw</a> a 
     "{{NotFoundError!!exception}}" {{DOMException}}.
    </ol>
```
This will require changing the HTML Standard to export "concept-submit-button" and "form owner" ("form owner" maybe also needs a suitable for attribute value).

Good call on using a `TypeError` and making that check first. That allows some flexibility between enforcing that through Web IDL and an algorithm.